cli/src/pcluster/models/cluster_resources.py

```diff
@@ -74,6 +74,22 @@ def batch_compute_environment(self):
         """Return Batch compute environment."""
         return self._get_output("BatchComputeEnvironmentArn")
 
+    def get_failure_reason(self):
+        """Reason of the failure when the cluster_status is in CREATE_FAILED."""
+        stack_events = get_all_stack_events(self.name)
+        wait_conditiaon_failure_events = None
+        for chuck in stack_events:
+            if wait_conditiaon_failure_events:
+                break
+            for event in chuck:
+                if (
+                    event.get("ResourceType") == "AWS::CloudFormation::WaitCondition"
```

Comment:
I would break out this predicate so that the logic for determining the correct event is separate (as a local function) which will provide better self-documenting code

Reply:
If you break out the predicate you can use the following to functionally find the first match:

```
found = next(filter(is_failed_wait, chunk), None)
…
```

Reply:
this is a good idea! Changed